tests/irdl/test_declarative_assembly_format.py

```diff
@@ -1312,23 +1304,66 @@ class OptionalRegionOp(IRDLOperation):
 
 
 def test_multiple_optional_regions():
-    """Test the parsing of multiple optional regions"""
-
-    """Test that multiple optional regions requires the ABCMeta PyRDL option."""
+    """Test that a variadic region variable cannot directly follow another variadic region variable."""
     with pytest.raises(
         PyRDLOpDefinitionError,
-        match="Operation test.optional_regions defines more than two variadic regions",
+        match="A variadic region variable cannot be followed by another variadic region variable.",
     ):
 
         @irdl_op_definition
-        class OptionalOperandsOp(IRDLOperation):  # pyright: ignore[reportUnusedClass]
+        class OptionalRegionsOp(IRDLOperation):  # pyright: ignore[reportUnusedClass]
             name = "test.optional_regions"
+            irdl_options = [AttrSizedRegionSegments()]
             region1 = opt_region_def()
             region2 = opt_region_def()
 
             assembly_format = "attr-dict-with-keyword $region1 $region2"
 
 
+@pytest.mark.parametrize(
+    "format, program, generic_program",
+    [
+        (
+            "($opt_region^ `keyword`)? attr-dict",
+            "test.optional_region_group",
+            '"test.optional_region_group"() : () -> ()',
+        ),
+        (
+            "($opt_region^ `keyword`)? attr-dict",
+            'test.optional_region_group {\n  "test.op"() : () -> ()\n} keyword',
+            '"test.optional_region_group"() ({"test.op"() : () -> ()}) : () -> ()',
+        ),
+        (
+            "(`keyword` $opt_region^)? attr-dict",
+            "test.optional_region_group",
+            '"test.optional_region_group"() : () -> ()',
+        ),
+        (
+            "(`keyword` $opt_region^)? attr-dict",
+            'test.optional_region_group keyword {\n  "test.op"() : () -> ()\n}',
+            '"test.optional_region_group"() ({ "test.op"() : () -> ()}) : () -> ()',
+        ),
+    ],
+)
+def test_optional_groups_regions(format: str, program: str, generic_program: str):
+    """Test the parsing of optional regions in an optional group"""
+
+    @irdl_op_definition
+    class OptionalRegionOp(IRDLOperation):  # pyright: ignore[reportUnusedClass]
```

Comment:
Isn't it used below?
```suggestion
    class OptionalRegionOp(IRDLOperation):
```